packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -317,38 +356,46 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
                 title: tooltip
             };
         }
-        const highlight = this.getDecorationData(node, 'highlight')[0];
         const children: React.ReactNode[] = [];
         const caption = node.name;
+        const highlight = this.getDecorationData(node, 'highlight')[0];
         if (highlight) {
-            let style: React.CSSProperties = {};
-            if (highlight.color) {
-                style = {
-                    ...style,
-                    color: highlight.color
-                };
-            }
-            if (highlight.backgroundColor) {
-                style = {
-                    ...style,
-                    backgroundColor: highlight.backgroundColor
-                };
-            }
-            const createChildren = (fragment: TreeDecoration.CaptionHighlight.Fragment) => {
-                const { data } = fragment;
-                if (fragment.highligh) {
-                    return <mark className={TreeDecoration.Styles.CAPTION_HIGHLIGHT_CLASS} style={style}>{data}</mark>;
-                } else {
-                    return data;
-                }
-            };
-            children.push(...TreeDecoration.CaptionHighlight.split(caption, highlight).map(createChildren));
-        } else {
+            children.push(this.toReactNode(caption, highlight));
+        }
+        const searchHighlight = this.searchHighlights ? this.searchHighlights.get(node.id) : undefined;
+        if (searchHighlight) {
+            children.push(...this.toReactNode(caption, searchHighlight));
+        } else if (!highlight) {
             children.push(caption);
         }
         return React.createElement('div', attrs, ...children);
     }
 
+    private toReactNode(caption: string, highlight1: TreeDecoration.CaptionHighlight): React.ReactNode[] {
```

Comment:
`protected` + `highlight1` should be `highlight`.

Reply:
fixed